### PR TITLE
[three] Fix type definition of WebXRManager

### PR DESCRIPTION
**Pull request by 0b5vr**
~~Waiting for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50930 to be merged~~ Ready

Update type definition of WebXRManager.
The primary motivation is to follow changes made in https://github.com/mrdoob/three.js/pull/20754.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
    - I have tried to use this changes in our private project and it's working good.
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
    - Add a test: `webxr-vr-cube`
        - I've found an issue around `XRSession` because of this test, so I also have fixed this
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mrdoob/three.js/blob/master/src/renderers/webxr/WebXRManager.js
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/three/src/core/Object3D.d.ts [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+    setRotationFromEuler(euler: Euler): void;
+
+    setRotationFromMatrix(m: Matrix4): void;
+
+    setRotationFromQuaternion(q: Quaternion): void;
+
+    /**
+     * Rotate an object along an axis in object space. The axis is assumed to be normalized.
+     * @param axis	A normalized vector in object space.
```
Comment (danger-public) [inline types/three/src/core/Object3D.d.ts:229]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by danger-public** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/three/examples/jsm/modifiers/EdgeSplitModifier.d.ts#L7** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  **types/three/src/loaders/Loader.d.ts#L36** - The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.
  </td>
    </tr>
  </tbody>
</table>



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## three ([<kbd>unpkg</kbd>](https://unpkg.com/browse/three@latest/))
was missing the following properties:
1. wrap
1. isGeneratorFunction
1. mark
1. awrap
1. AsyncIterator

<details>
<summary>as well as these 35 other properties...</summary>
<p>async, keys, values, AxisHelper, BinaryTextureLoader, BoundingBoxHelper, CanvasRenderer, ClosedSplineCurve3, CustomToneMapping, DynamicBufferAttribute, EdgesHelper, Face3, Face4, FaceColors, JSONLoader, LensFlare, Math, MeshFaceMaterial, MultiMaterial, NoColors, Particle, ParticleBasicMaterial, ParticleSystem, ParticleSystemMaterial, PointCloud, PointCloudMaterial, SceneUtils, Spline, SplineCurve3, Vertex, VertexColors, WebGLRenderTargetCube, WebGLUtils, WireframeHelper, XHRLoader</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d4077c8f3c146f0b4d5f1be7c355ec1539305b19
</p>

**Comment by FMS-Cat** [comment]
ohhhh it notifies to previous contributors of the package. that's actually cool

**Comment by joshuaellis** [comment]
@FMS-Cat Thanks for the input, the core repo for these types is [three-ts-types](https://github.com/three-types/three-ts-types). I'm happy to move them over to that repo after we merge here & add you to 'all-contributors' if you'd like? 

Alternatively, you can move the PR. I don't mind personally.

**Comment by FMS-Cat** [comment]
oh. that sounds much easier to contribute to. I'm 100% ready to move the PR to the three-ts-types repo if you want to use that repo as a single source of truth.

**Comment by joshuaellis** [comment]
It should be a lot easier yes! They're still hosted on `DT` and published under the namespace `@types/` but we felt it was easier for everyone if we did it that way.

**Comment by FMS-Cat** [comment]
Sure! Made a PR: https://github.com/three-types/three-ts-types/pull/7 .
Should I close this one for now?

**Comment by joshuaellis** [comment]
You should 👍🏼

**Pull request closed without merging**